[jit] remove FunctionType as an allowed constant

suo opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#29405 [jit] remove FunctionType as an allowed constant**

We never actually used this (function attributes are a separate pathway
in ConcreteModuleType).

Differential Revision: [D18378392](https://our.internmc.facebook.com/intern/diff/D18378392)

suo committed:
[jit] remove FunctionType as an allowed constant

We never actually used this (function attributes are a separate pathway
in ConcreteModuleType).

suo committed:
Update on "[jit] remove FunctionType as an allowed constant"


We never actually used this (function attributes are a separate pathway
in ConcreteModuleType).

Differential Revision: [D18378392](https://our.internmc.facebook.com/intern/diff/D18378392)

kostmo commented:
## CircleCI build failures summary
As of commit eaaf715:
* 0/1 recognized as flaky
* 1/1 failures introduced in this PR



Here are the [reasons each build failed](https://dr.pytorch.org/commit-details.html?sha1=eaaf715b6a8c0eda6fbfcfc4b2c325bac0ef74a0).

---
This comment was automatically generated by [Dr. CI](https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment).
Follow [this link to opt-out](https://dr.pytorch.org/admin/comments-opt-out.html) of these comments for your Pull Requests.


Please report bugs/suggestions on the [GitHub issue tracker](https://github.com/kostmo/circleci-failure-tracker/issues).

suo committed:
Update on "[jit] remove FunctionType as an allowed constant"


We never actually used this (function attributes are a separate pathway
in ConcreteModuleType).

Differential Revision: [D18378392](https://our.internmc.facebook.com/intern/diff/D18378392)
